pkg/kubelet/kubelet.go

```diff
@@ -3674,3 +3707,60 @@ func (kl *Kubelet) shapingEnabled() bool {
 	}
 	return true
 }
+
+// killPodNow kills a pod.
+// The pod status is updated, and then it is killed with the specified grace period.
+// This function blocks until the pod is killed, an error is encountered, or a timeout is determined.
```

Comment:
pick a different pod for the first iteration.  The cleanup routine will also try to kill the pod but could likely error for the same underlying reason (block on disk or network io maybe).  Killing an extra pod seems safest for now.  As for the race between cleanup and this routine, I think it's small and the net would be redundant calls to docker stop or systemctl stop, and the shortest timeout period specified should win meaning eviction with 0 should win most races.